plugin/storage/es/spanstore/reader.go

```diff
@@ -209,20 +218,23 @@ func bucketToStringArray(buckets []*elastic.AggregationBucketKeyItem) ([]string,
 
 // FindTraces retrieves traces that match the traceQuery
 func (s *SpanReader) FindTraces(ctx context.Context, traceQuery *spanstore.TraceQueryParameters) ([]*model.Trace, error) {
+	span, ctx := startSpanForQuery(ctx, "FindTraces", traceQuery.ServiceName)
+	defer span.Finish()
+
 	if err := validateQuery(traceQuery); err != nil {
 		return nil, err
 	}
 	if traceQuery.NumTraces == 0 {
 		traceQuery.NumTraces = defaultNumTraces
 	}
-	uniqueTraceIDs, err := s.findTraceIDs(traceQuery)
+	uniqueTraceIDs, err := s.findTraceIDs(span, traceQuery)
 	if err != nil {
 		return nil, err
 	}
-	return s.multiRead(uniqueTraceIDs, traceQuery.StartTimeMin, traceQuery.StartTimeMax)
+	return s.multiRead(span, uniqueTraceIDs, traceQuery.StartTimeMin, traceQuery.StartTimeMax)
 }
 
-func (s *SpanReader) multiRead(traceIDs []string, startTime, endTime time.Time) ([]*model.Trace, error) {
+func (s *SpanReader) multiRead(span opentracing.Span, traceIDs []string, startTime, endTime time.Time) ([]*model.Trace, error) {
```

Comment:
It is non-idiomatic to pass spans around. Can you pass Context instead?

Reply:
Sure. 

Reason I did it was that I was working with #1038 as a reference. Span was passed to a specific function to log any errors that occurred. 
https://github.com/jaegertracing/jaeger/pull/1038/files#diff-74fbf0ea62bbce4e040d3499e9728171R386

Anyway, since `GetTrace` has just one ES-querying-function, it is OK to treat the span of `GetTrace` as representative for `multiRead`?

Reply:
My bad then, I was lazy. We can always extract the span from the context.

Regarding multi-read, if having more instrumentation can be helpful in debugging, then it's better to keep extra data.

Reply:
Okay, sure! Working on this. :)